CWE-94-Python_darwin/andy897221_Proof-of-Play-Flow-Demo/PoP.py

```
@@ -0,0 +1,107 @@
+    def broadcast_gameRec(self):
+        return requests.post(f"http://127.0.0.1:{self.api_port}/broadcast").text
+
+    def _direct_send_match(self, data):
+        requests.post(f'http://127.0.0.1:{self.blockchain_port}/matches/new', data=pickle.dumps(data))
+        return
+
+    def return_plyrList(self):
+        return pickle.loads(requests.get(f"http://127.0.0.1:{self.api_port}/plyrList").content)
+
+    def return_chain_status(self):
+        return pickle.loads(requests.get(f"http://127.0.0.1:{self.blockchain_port}/chain/status").content)
```

Ruff (0.13.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)